admin/app/controllers/authenticated/trainings/new.js

```diff
@@ -0,0 +1,40 @@
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import Controller from '@ember/controller';
+
+export default class NewController extends Controller {
+  @service notifications;
+  @service store;
+  @service router;
+
+  @action
+  goBackToTrainingList() {
+    this.router.transitionTo('authenticated.trainings.list');
+  }
+
+  @action
+  async createTraining(event) {
+    event.preventDefault();
+    const training = this.model;
+
+    try {
+      await training.save();
+      this.notifications.success('Le contenu formatif a été créé avec succès.');
+      this.router.transitionTo('authenticated.trainings.list');
+    } catch (error) {
+      this._handleResponseError(error);
+    }
+  }
+
+  _handleResponseError({ errors }) {
+    if (!errors) {
+      return this.notifications.error('Une erreur est survenue.');
+    }
+    errors.forEach((error) => {
+      if (['404', '412', '422'].includes(error.status)) {
+        return this.notifications.error(error.detail);
+      }
+      return this.notifications.error('Une erreur est survenue.');
+    });
+  }
```

Comment:
question : Notre endpoint pour créér les trainings peut renvoyer des 404/412/422 ?

Je propose de simplifier le traitement d'erreur : 
```suggestion
      this.notifications.error('Une erreur est survenue.');
    }
  }
```